syntaxes/src/types.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export type GrammarDefinitionValue = string|RegExp|GrammarDefinition|GrammarDefinition[];
+
+export interface GrammarDefinition {
+  [key: string]: GrammarDefinitionValue;
+}
+
+export type JsonObjectValue = string|JsonObject|JsonObject[];
+
+export interface JsonObject {
+  [key: string]: JsonObjectValue;
```

Comment:
Consider

```suggestion
  [key: string]: string|JsonObject|JsonObject[];
```

and elision of the type on line 15

Reply:
Fixed!